lib/schema_plus/indexes.rb

```diff
@@ -1,12 +1,19 @@
 require 'schema_plus/core'
 require 'its-it'
+require 'active_record'
+
+if Gem::Version.new(ActiveRecord::VERSION::STRING) >= Gem::Version.new('5.2.0')
+  require_relative 'indexes/remove_if_exists_5_2'
```

Comment:
I'm OK with this, but maybe @ronen knows offhand if there's another convention that the gems are following?

Reply:
Another approach would be to do this checking in the remove_if_exists / index_definition rb files directly and condition the logic there. e.g.
```ruby
if Gem::Version.new(.....) >= ....
  def remove_if_exists(table_name, *args)
    ...
  end
else
  def remove_if_exists(table_name, *args)
    ...
  end
end
```

Or just do the check in the one method

Reply:
I would define the methods conditionally like you did, so we don't run the version checks every time a method is called. That could be micro-optimising, but I think people are complaining about Ruby and Rails being slow enough 😄

Reply:
> maybe @ronen knows offhand if there's another convention that the gems are following?

Nothing jumps to mind.   I've sometimes wondered if schema_monkey should have a mechanism for version-specific module inclusion, just like it has for db-specific module inclusion.  But I'm not exactly sure what it would look like.